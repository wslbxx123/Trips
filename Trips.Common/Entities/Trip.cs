using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;

namespace Trips.Core.Entities
{
    public class Trip
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        [BsonElement("name")]
        public string Name { get; set; }
        [BsonElement("description")]
        public string Description { get; set; }
        [BsonElement("dateStarted")]
        public DateTime DateStarted { get; set; }
        [BsonElement("dateCompleted")]
        public DateTime? DateCompleted { get; set; }
    }
}
