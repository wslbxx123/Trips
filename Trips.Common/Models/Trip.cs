using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Text;

namespace Trips.Common.Models
{
    public class Trip
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        [BsonElement("name")]
        public string Name { get; set; }
        [BsonElement("description")]
        public string Description { get; set; }
        [BsonElement("dateStarted")]
        public DateTime DateStarted { get; set; }
        [BsonElement("dateCompleted")]
        public DateTime? DateCompleted { get; set; }
    }
}
