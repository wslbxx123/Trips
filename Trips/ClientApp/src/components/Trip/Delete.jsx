import React, {useState, useEffect} from 'react';
import axios from 'axios';

export function Delete(props) {
    const [name, setName] = useState('');
    const [description, setDescription] = useState('');
    const [dateStarted, setDateStarted] = useState(null);
    const [dateCompleted, setDateCompleted] = useState(null);

    useEffect(() => {
        const {id} = props.match.params;

        axios.get("api/Trips/GetTrip/"+id).then(trip => {
            const response = trip.data;

            setName(response.name);
            setDescription(response.description);
            setDateStarted(new Date(response.dateStarted).toISOString().slice(0,10));
            setDateCompleted(response.dateCompleted ? new Date(response.dateCompleted).toISOString().slice(0, 10) : null);
        });
    });

    function onCancel(e) {
        const {history} = this.props;
    
        history.push("/trips");
    }
    
    function onConfirmation(e) {
        const {id} = props.match.params;
        const {history} = props;
    
        axios.delete("/api/Trips/DeleteTrip/" + id).then(result => {
            history.push("/trips");
        });
    }

    return (
        <div style={{marginTop: 10}}>
            <h2>Delete trip confirmation</h2>
            <div className="card">
                <div className="card-body">
                    <h4 className="card-title">{name}</h4>
                    <p className="card-text">{description}</p>
                    <button onClick={onCancel} className="btn btn-default">
                        Cancel
                    </button>
                    <button onClick={onConfirmation} className="btn btn-danger">
                        Confirm
                    </button>
                </div>
            </div>
        </div>
    ); 
}