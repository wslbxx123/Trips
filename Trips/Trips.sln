
Microsoft Visual Studio Solution File, Format Version 12.00
# Visual Studio Version 16
VisualStudioVersion = 16.0.29709.97
MinimumVisualStudioVersion = 10.0.40219.1
Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Trips", "Trips.csproj", "{6AA91CF0-F914-47ED-ACDB-8511DFC5158A}"
EndProject
Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Trips.Application", "..\Trips.Application\Trips.Application.csproj", "{856CFBC7-1924-4746-8DA1-48E31F5AEDC0}"
EndProject
Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Trips.Core", "..\Trips.Core\Trips.Core.csproj", "{2E7A5169-954E-4280-82CA-521632B47F36}"
EndProject
Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Trips.Infrastructure", "..\Trips.Infrastructure\Trips.Infrastructure.csproj", "{E2F5B798-6854-439C-A6D1-F05AA096C134}"
EndProject
Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Trips.Application.UnitTests", "..\Trips.Application.UnitTests\Trips.Application.UnitTests.csproj", "{A12A6BA6-6667-4E39-BD57-8DE3B532D91A}"
EndProject
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Trips.Infrastructure.UnitTests", "..\Trips.Infrastructure.UnitTests\Trips.Infrastructure.UnitTests.csproj", "{DD122DB3-0556-4CF5-82FB-B99FE3F10063}"
EndProject
Global
	GlobalSection(SolutionConfigurationPlatforms) = preSolution
		Debug|Any CPU = Debug|Any CPU
		Release|Any CPU = Release|Any CPU
	EndGlobalSection
	GlobalSection(ProjectConfigurationPlatforms) = postSolution
		{6AA91CF0-F914-47ED-ACDB-8511DFC5158A}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{6AA91CF0-F914-47ED-ACDB-8511DFC5158A}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{6AA91CF0-F914-47ED-ACDB-8511DFC5158A}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{6AA91CF0-F914-47ED-ACDB-8511DFC5158A}.Release|Any CPU.Build.0 = Release|Any CPU
		{856CFBC7-1924-4746-8DA1-48E31F5AEDC0}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{856CFBC7-1924-4746-8DA1-48E31F5AEDC0}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{856CFBC7-1924-4746-8DA1-48E31F5AEDC0}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{856CFBC7-1924-4746-8DA1-48E31F5AEDC0}.Release|Any CPU.Build.0 = Release|Any CPU
		{2E7A5169-954E-4280-82CA-521632B47F36}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{2E7A5169-954E-4280-82CA-521632B47F36}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{2E7A5169-954E-4280-82CA-521632B47F36}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{2E7A5169-954E-4280-82CA-521632B47F36}.Release|Any CPU.Build.0 = Release|Any CPU
		{E2F5B798-6854-439C-A6D1-F05AA096C134}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{E2F5B798-6854-439C-A6D1-F05AA096C134}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{E2F5B798-6854-439C-A6D1-F05AA096C134}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{E2F5B798-6854-439C-A6D1-F05AA096C134}.Release|Any CPU.Build.0 = Release|Any CPU
		{A12A6BA6-6667-4E39-BD57-8DE3B532D91A}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{A12A6BA6-6667-4E39-BD57-8DE3B532D91A}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{A12A6BA6-6667-4E39-BD57-8DE3B532D91A}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{A12A6BA6-6667-4E39-BD57-8DE3B532D91A}.Release|Any CPU.Build.0 = Release|Any CPU
		{DD122DB3-0556-4CF5-82FB-B99FE3F10063}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{DD122DB3-0556-4CF5-82FB-B99FE3F10063}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{DD122DB3-0556-4CF5-82FB-B99FE3F10063}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{DD122DB3-0556-4CF5-82FB-B99FE3F10063}.Release|Any CPU.Build.0 = Release|Any CPU
	EndGlobalSection
	GlobalSection(SolutionProperties) = preSolution
		HideSolutionNode = FALSE
	EndGlobalSection
	GlobalSection(ExtensibilityGlobals) = postSolution
		SolutionGuid = {E21732DD-68B2-454B-BC15-7A83B1436541}
	EndGlobalSection
EndGlobal
