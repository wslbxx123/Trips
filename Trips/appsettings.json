{
  "TripDatabaseSettings": {
    "TripsCollectionName": "Trips",
    "ConnectionString": "mongodb://localhost:27017",
    "DatabaseName": "TripDb"
  },
  "Logging": {
    "LogLevel": {
      "Default": "Information",
      "Microsoft": "Warning",
      "Microsoft.Hosting.Lifetime": "Information"
    }
  },
  "AllowedHosts": "*"
}
