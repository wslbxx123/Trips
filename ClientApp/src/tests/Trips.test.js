import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { Trips } from '../components/Trip/Trips'

const { mount } = Enzyme

Enzyme.configure({ adapter: new Adapter() })

function setup() {
  const props = {
    getAllTrips: jest.fn(),
    trips: {
      loading: false
    }
  };

  const enzymeWrapper = mount(<Trips {...props}/>);

  return {
    props,
    enzymeWrapper
  };
}

describe('<Trips />', function () {
  it('render correctly', () => {
    const {wrapper, props} = setup()
    const getSpy = jest.spyOn(props, 'getAllTrips');
    expect(getSpy).toBeCalled();
  })
})