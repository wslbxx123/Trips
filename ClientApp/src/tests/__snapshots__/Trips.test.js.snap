// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load and display as snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      All trips
    </h1>
    <p>
      Here you can see all trips
    </p>
    0
  </div>
</DocumentFragment>
`;
